### PR TITLE
xsdk: add amrex variant

### PR DESCRIPTION
**Pull request by balay**
and remove compiler conditionals [as amrex conflict with clang no longer exists since #22967]

### CODE REVIEW

**Review by v-dobrev — APPROVED** [review]

File: var/spack/repos/builtin/packages/xsdk/package.py [hunk 1]
```diff
@@ -207,12 +208,8 @@ class Xsdk(BundlePackage, CudaPackage, ROCmPackage):
     xsdk_depends_on("magma@2.7.0", when="@0.8.0", cuda_var="?cuda", rocm_var="?rocm")
 
     xsdk_depends_on("amrex +sycl", when="@1.0.0: +sycl")
-    xsdk_depends_on("amrex@23.08+sundials", when="@1.0.0 %intel", cuda_var="cuda", rocm_var="rocm")
-    xsdk_depends_on("amrex@23.08+sundials", when="@1.0.0 %gcc", cuda_var="cuda", rocm_var="rocm")
-    xsdk_depends_on("amrex@23.08+sundials", when="@1.0.0 %cce", cuda_var="cuda", rocm_var="rocm")
-    xsdk_depends_on("amrex@22.09+sundials", when="@0.8.0 %intel", cuda_var="cuda", rocm_var="rocm")
-    xsdk_depends_on("amrex@22.09+sundials", when="@0.8.0 %gcc", cuda_var="cuda", rocm_var="rocm")
-    xsdk_depends_on("amrex@22.09+sundials", when="@0.8.0 %cce", cuda_var="cuda", rocm_var="rocm")
+    xsdk_depends_on("amrex@23.08+sundials", when="@1.0.0 +amrex", cuda_var="cuda", rocm_var="rocm")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/xsdk/package.py:211]: These functions are very hard to follow, wish we could unwrap them someday.

**Review by alalazo — APPROVED** [review]
